lib/runtime/src/utils/tasks/tracker.rs

```diff
@@ -0,0 +1,6285 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! # Task Tracker - Hierarchical Task Management System
```

Comment:
For reviewers - you can run something like `cargo doc` on these changes to review all the examples in browser with syntax highlighting and such